core/src/test/java/io/parsingdata/metal/AutoEqualityTest.java

```diff
@@ -361,32 +355,36 @@ private static List<Object> instantiate(final List<Supplier<Object>> suppliers)
         return output;
     }
 
-    @Test
-    public void notEqualsNull() {
+    @ParameterizedTest
+    @MethodSource("data")
+    public void notEqualsNull(final Object object, final Object same, final Object[] other) {
         assertFalse(object.equals(null));
```

Comment:
IntellIJ IDEA suggests `assertNotEquals(null, object);` here (and in a few other places in this test class, `ConstantSliceTest` and `EqualityTest`).

Reply:
The `equals` method is the method being tested here, so it is deliberate that the test is written like this. We want to ignore any other checks that is possibly checked when calling `assertNotEquals`. This is the test that really checks if the `equals` method returns a result as expected, so I am going to keep it as is.